Decompiler/Decompiler.cs

```diff
@@ -412,11 +412,13 @@ private void ProcessFile(string path, Stream stream, string originalPath = null)
 
                 if (OutputFile != null)
                 {
-                    var data = FileExtract.Extract(resource);
+                    var contentFile = FileExtract.Extract(resource);
 
                     var filePath = Path.ChangeExtension(path, extension);
 
-                    DumpFile(filePath, data);
+                    // TODO: fix exporting subfiles to a non-folder --output
```

Comment:
How hard would it be to fix? What currently happens?

Reply:
> What currently happens?

Currently any *possible* subfiles are skipped. Vpk dumping & folder dumping works well but there is an issue when doing 

`decompiler -i "sprite.vtex_c" -o "sprite.png" -d`

DumpFile has some busted `filePath` logic and sends the last subfile to `sprite.png` as well, instead of the subfile path that I'm giving to it. I just can't fix it in a nice way.

Reply:
I wouldn't say it's busted given the restrictions we had before. I'd say one way of fixing it would be to dump the 'main' file to the given output path as-is, and for subfiles either name them similarly to the output path (but append number to it or whatever), or force useOutputAsDirectory to true.

Reply:
I would like to deal with this after we've got some subfiles rolling. Decompiler is very folder/pak oriented and it's hard not to break those uses in the process.

Reply:
If I were to merge this pr as-is, will it break current scripts?

How do we avoid the breakage?

Reply:
Yes, works as before and vpk and folder decompiling will create extra files where available.

> How do we avoid the breakage?

Imo it shouldn't be allowed to put a file as -o. The code can be much cleaner then.

Reply:
Nevermind. Did a slight refactor to the original and fixed it.